core/types/block.libevm.go

```diff
@@ -27,40 +28,50 @@ import (
 // HeaderHooks are required for all types registered with [RegisterExtras] for
 // [Header] payloads.
 type HeaderHooks interface {
+	MarshalJSON(*Header) ([]byte, error) //nolint:govet // Type-specific override hook
+	UnmarshalJSON(*Header, []byte) error //nolint:govet
 	EncodeRLP(*Header, io.Writer) error
 	DecodeRLP(*Header, *rlp.Stream) error
 }
 
+// hooks returns the Header's registered HeaderHooks, if any, otherwise a
+// [NOOPHeaderHooks] suitable for running default behaviour.
+func (h *Header) hooks() HeaderHooks {
+	if r := registeredExtras; r.Registered() {
+		return r.Get().hooks.hooksFromHeader(h)
+	}
```

Comment:
nit no need to assign to `r`:

```suggestion
	if registeredExtras.Registered() {
		return registeredExtras.Get().hooks.hooksFromHeader(h)
	}
```

Reply:
I did it to avoid unnecessary verbosity and code stuttering. One of the core guiding principles that I follow (and recommend) is for code to get out of the way of the human seeing the meaning being communicated to the computer. Here there's a trade-off between distraction (a 5-syllable variable being unnecessarily repeated, with 3 of them repeated again in the method name) and a meat-RAM allocation (remembering what `r` is). I chose the latter because it only needs to be remembered for one line.

Reply:
I was actually saying this for readability more than for memory usage 😄 
Assigning to `r` made me think initially we were copying the variable because the method calls were perhaps modifying it. I feel it could hide something eventually if you see what I mean?